File: delete_editorcard.swift
import Foundation
let path = "Promtier/Views/NewPromptView.swift"
let content = try String(contentsOfFile: path, encoding: .utf8)
var lines = content.components(separatedBy: .newlines)
let startIdx = lines.firstIndex { $0.hasPrefix("struct EditorCard: View {") }
if let s = startIdx {
    var endIdx = s
    var braceCount = 0
    for i in s..<lines.count {
        braceCount += lines[i].components(separatedBy: "{").count - 1
        braceCount -= lines[i].components(separatedBy: "}").count - 1
        if braceCount == 0 { endIdx = i; break }
    }
    lines.removeSubrange(s...endIdx)
    try lines.joined(separator: "\n").write(toFile: path, atomically: true, encoding: .utf8)
    print("Deleted EditorCard")
}

File: delete_editorcard.rb
require 'fileutils'
content = File.read("Promtier/Views/NewPromptView.swift")
lines = content.lines

start_idx = lines.index { |l| l.include?("struct EditorCard: View {") }
if start_idx
  brace_count = 0
  end_idx = start_idx
  lines[start_idx..-1].each_with_index do |line, i|
    brace_count += line.count('{')
    brace_count -= line.count('}')
    if brace_count == 0
      end_idx = start_idx + i
      break
    end
  end
  lines.slice!(start_idx..end_idx)
  File.write("Promtier/Views/NewPromptView.swift", lines.join)
  puts "Deleted EditorCard length #{end_idx - start_idx + 1}"
end
